jerry-core/api/jerry-snapshot.c

```diff
@@ -997,16 +1001,68 @@ jerry_snapshot_result (const uint32_t *snapshot_p, /**< snapshot */
   }
   else
   {
+    ecma_value_t user_value = ECMA_VALUE_EMPTY;
+
+    if (options_p != NULL
+        && (options_p->options & JERRY_PARSE_HAS_USER_VALUE))
+    {
+      user_value = options_p->user_value;
+    }
+
+    uint32_t script_size = (user_value != ECMA_VALUE_EMPTY ? sizeof (cbc_script_user_t)
+                                                           : sizeof (cbc_script_t));
+    cbc_script_t *script_p = jmem_heap_alloc_block (script_size);
+
+    script_p->refs_and_type = (CBC_SCRIPT_REF_ONE | CBC_SCRIPT_GENERIC);
+
+    if (user_value != ECMA_VALUE_EMPTY)
+    {
+      script_p->refs_and_type = (CBC_SCRIPT_REF_ONE | CBC_SCRIPT_USER_VALUE);
+
+      if (ecma_is_value_object (user_value))
+      {
+        script_p->refs_and_type = (CBC_SCRIPT_REF_ONE | CBC_SCRIPT_USER_OBJECT);
+      }
+    }
```

Comment:
This code part seems very similar to the changes in the parser (see: https://github.com/jerryscript-project/jerryscript/pull/4710/files#diff-baabef101cbc924f0aff83bdc8e60e17890ed01390e7fcc9fd519c8b07bb4171R1846 ). Maybe we could have a common method for this?

Reply:
Similar but not the same. I think it is not worth for combining it. If you insist, where should we put the function? Into the parser or into the snapshot?

Reply:
I have added a macro for it.